lib/text/ssa_text_parser.js

```diff
@@ -0,0 +1,352 @@
+      cue.letterSpacing = letterSpacing + 'px';
+    }
+    const alignment = style['Alignment'];
+    if (alignment) {
+      const alignmentInt = parseInt(alignment, 10);
+      switch (alignmentInt) {
+        case 1:
+          cue.displayAlign = Cue.displayAlign.AFTER;
+          cue.textAlign = Cue.textAlign.START;
+          break;
+        case 2:
+          cue.displayAlign = Cue.displayAlign.AFTER;
+          cue.textAlign = Cue.textAlign.CENTER;
+          break;
+        case 3:
+          cue.displayAlign = Cue.displayAlign.AFTER;
+          cue.textAlign = Cue.textAlign.END;
+          break;
+        case 5:
+          cue.displayAlign = Cue.displayAlign.BEFORE;
+          cue.textAlign = Cue.textAlign.START;
+          break;
+        case 6:
+          cue.displayAlign = Cue.displayAlign.BEFORE;
+          cue.textAlign = Cue.textAlign.CENTER;
+          break;
+        case 7:
+          cue.displayAlign = Cue.displayAlign.BEFORE;
+          cue.textAlign = Cue.textAlign.END;
+          break;
+        case 9:
+          cue.displayAlign = Cue.displayAlign.CENTER;
+          cue.textAlign = Cue.textAlign.START;
+          break;
+        case 10:
+          cue.displayAlign = Cue.displayAlign.CENTER;
+          cue.textAlign = Cue.textAlign.CENTER;
+          break;
+        case 11:
+          cue.displayAlign = Cue.displayAlign.CENTER;
+          cue.textAlign = Cue.textAlign.END;
+          break;
+      }
+    }
+    const opacity = style['AlphaLevel'];
+    if (opacity) {
+      cue.opacity = parseFloat(opacity);
+    }
+  }
+
+  /**
+   * Parses a SSA color .
+   *
+   * @param {string} colorString
+   * @return {?string}
+   * @private
+   */
+  static parseSsaColor_(colorString) {
+    // The SSA V4+ color can be represented in hex (&HAABBGGRR) or in decimal
+    // format (byte order AABBGGRR) and in both cases the alpha channel's
+    // value needs to be inverted as in case of SSA the 0xFF alpha value means
+    // transparent and 0x00 means opaque
+    /** @type {number} */
+    let abgr = -1;
+    try {
+      if (colorString.startsWith('&H')) {
+        // See: https://bugs.chromium.org/p/v8/issues/detail?id=2869
+        let abgrString = (' ' + colorString).slice(1);
+        if (colorString.length < 10) {
+          // Add leading zeros if necessary.
+          while (abgrString.length != 10) {
+            abgrString = [
+              abgrString.slice(0, 2),
+              '0',
+              abgrString.slice(2),
+            ].join('');
+          }
+        }
+        abgr = parseInt(abgrString.substring(2), 16);
+      } else {
+        // Parse color from decimal format (bytes order AABBGGRR).
+        abgr = parseInt(colorString, 16);
+      }
+    } catch (e) {
+      shaka.log.warning('Error parsing SSA color.', colorString);
+    }
+    if (abgr >= 0) {
+      const a = ((abgr >> 24) & 0xFF) ^ 0xFF; // Flip alpha.
+      let alpha = Math.min(1, Math.max(0, (a / 255).toFixed(2)));
```

Comment:
`ToFixed` creates a string, better to keep it a number here.  Also, the math above seems to prohibit outside range values, so the min/max doesn't seem needed.

Reply:
I changed it!